File: backend/logsheet/src/main/java/com/sunbeam/logsheet/service/RoleService.java
package com.sunbeam.logsheet.service;

import com.sunbeam.logsheet.DTO.*;

import java.util.List;

public interface RoleService {
    ApiResponse<?> createRole(RoleCreateDTO dto);
    ApiResponse<?> updateRole(Long id, RoleUpdateDTO dto);
    ApiResponse<?> deleteRole(Long id);
    RoleResponseDTO getRoleById(Long id);
    List<RoleResponseDTO> getAllRoles();
    List<RoleResponseDTO> getRolesByTitleDTO(String title);
}

File: frontend/logsheet_fe/src/services/roleService.js
import api from "../utils/api"; 


export const createRole = (data) => api.post("/roles", data);


export const updateRole = (id, data) => api.put(`/roles/${id}`, data);


export const deleteRole = (id) => api.delete(`/roles/${id}`);


export const getRoleById = (id) => api.get(`/roles/${id}`);


export const getAllRoles = () => api.get("/roles");
